client/src/controllers/PreviewController.ts

```diff
@@ -0,0 +1,473 @@
+import { Controller } from '@hotwired/stimulus';
+import { WAGTAIL_CONFIG } from '../config/wagtailConfig';
+import { debounce } from '../utils/debounce';
+import { gettext } from '../utils/gettext';
+import type { ProgressController } from './ProgressController';
+
+/**
+ * Controls the preview panel component to submit the current form state and
+ * update the preview iframe if the form is valid.
+ */
+export class PreviewController extends Controller<HTMLElement> {
+  static classes = ['unavailable', 'hasErrors', 'selectedSize'];
+
+  static targets = [
+    'sizeInput',
+    'newTabButton',
+    'spinner',
+    'modeSelect',
+    'iframe',
```

Comment:
Just thinking it may be good to streamline these target names, removing the element type from the reference name + keeps it shorter. 

The fact that it's an input/button is probably a duplicate as the html elements that are the target will be these types. 

```
static targets = [
    'size',
    'newTab',
    'spinner',
    'mode',
    'iframe',
```

Also, not strict, but we have been sorting these alphabetically in other controllers.